### PR TITLE
[JENKINS-65161] remove digester 2 from core

### PR DESCRIPTION
**Pull request by olamy**

### CODE REVIEW

**Comment by olamy** [comment]
see https://github.com/jenkinsci/jenkins/pull/5320

File: src/main/java/hudson/plugins/dimensionsscm/DimensionsChangeLogParser.java [hunk 1]
```diff
@@ -59,8 +60,20 @@ public DimensionsChangeLogSet parse(final Run run, final RepositoryBrowser<?> br
...
+        digester.setXIncludeAware(false);
+        if (!Boolean.getBoolean(DimensionsChangeLogParser.class.getName() + ".UNSAFE")) {
+            try {
+                digester.setFeature("http://apache.org/xml/features/disallow-doctype-decl", true);
+                digester.setFeature("http://xml.org/sax/features/external-general-entities", false);
+                digester.setFeature("http://xml.org/sax/features/external-parameter-entities", false);
+                digester.setFeature("http://apache.org/xml/features/nonvalidating/load-external-dtd", false);
+            }
+            catch ( ParserConfigurationException ex) {
+                throw new SAXException("Failed to securely configure CVS changelog parser", ex);
```
Comment (dconneely) [inline src/main/java/hudson/plugins/dimensionsscm/DimensionsChangeLogParser.java:74]:
```suggestion
                throw new SAXException("Failed to securely configure Dimensions changelog parser", ex);
```

**Review by dconneely — APPROVED** [review]
LGTM, except for one minor suggestion I can fix later.

**Comment by olamy** [comment]
@dconneely ooops my bad sorry for the copy/paste typo :( 
Thanks for merging this!

**Comment by batmat** [comment]
@dconneely +1, thanks for merging this!

May I ask you whether you would consider releasing a version with this fix? We're trying to adapt the ecosystem to support https://github.com/jenkinsci/jenkins/pull/5320 so any release done is one plugin less to worry about :-).

Thanks a lot.

**Comment by dconneely** [comment]
@batmat @olamy Sure, I aim to cut a release in the next few days. Thanks for all the help with this!

**Comment by Wallace-EXE** [comment]
When using the latest Jenkins version with Dimensions plugin added (0.9.2) , I get the following error when building: `[DIMENSIONS] )
[DIMENSIONS] Calculating changes for directory '\'...
FATAL: org/apache/commons/digester/Digester
java.lang.NoClassDefFoundError: org/apache/commons/digester/Digester
	at hudson.plugins.dimensionsscm.DimensionsChangeLogParser.createDigester(DimensionsChangeLogParser.java:68)`
	
	It looks like Jenkins no longer has the Commons Digester Library: https://support.cloudbees.com/hc/en-us/articles/4402545935501--JEP-231-Commons-Digester-Library-Removal

**Comment by dconneely** [comment]
> When using the latest Jenkins version with Dimensions plugin added (0.9.2) , I get the following error when building:
> 
> `[DIMENSIONS] ) [DIMENSIONS] Calculating changes for directory '\'... FATAL: org/apache/commons/digester/Digester java.lang.NoClassDefFoundError: org/apache/commons/digester/Digester at hudson.plugins.dimensionsscm.DimensionsChangeLogParser.createDigester(DimensionsChangeLogParser.java:68)` It looks like Jenkins no longer has the Commons Digester Library: https://support.cloudbees.com/hc/en-us/articles/4402545935501--JEP-231-Commons-Digester-Library-Removal

Looks like you are correct and the wrong version of Apache Commons Digester was included in the POM w.r.t. the version used by the source code when this PR was merged. One temporary workaround in the meantime is to drop commons-digester-2.1.jar file into the plugins/dimensionsscm/WEB-INF/lib/ folder to make the older JAR available, but I'll aim to make a new release ASAP.

**Comment by olamy** [comment]
jar is ok.
```
 unzip -l target/dimensionsscm.hpi|grep digester
   241614  06-23-2021 18:14   WEB-INF/lib/commons-digester3-3.2.jar
```

But my bad import should be changed in DimensionsChangeLogParser to digester3 package.

**Comment by olamy** [comment]
see #41

**Comment by batmat** [comment]
@dconneely just to make maintenance simpler, now also that time has passed, I would recommend just bumping the `jenkins.version` for this plugin to a Jenkins Core version including the commons-digester removal. This way it will be more immediately visible is something is off.

That means either:
* [2.303.1 if you go for the first LTS with it](https://www.jenkins.io/changelog-stable/#v2.303.1), or
* [2.297 if you aim for the weekly](https://www.jenkins.io/changelog/#v2.297)

**Comment by dconneely** [comment]
Not currently able to do a release. I can use my Jenkins account (dconneely) to log in to accounts.jenkins.io, issues.jenkins.io, but it is not working with repo.jenkins-ci.org. Have tried changing my password on accounts.jenkins.io to no effect.

**Pull request merged**